### DISCUSSION TITLE
Can the parent field be accessed from the child class?

### DISCUSSION THREAD

**Discussion by yuki-kimoto** [Ovid/Cor#79] [Q&A]
4.2.2 Inheritance

https://github.com/Ovid/Cor/blob/master/rfc/classes.md#422-inheritance

Can the parent fields be accessed from the child class?

```
class Point {
  field $x;
  field $y;
}

class Point3D :isa(Point) {
  method foo () {
    # Can the parent field be accessed from the child class?
    print "$x\n";
  }
}
```

A python example: <code>tax</code>

```
class Animal():
    def __init__(self):
        self.tax = 1.1
class Dog(Animal):
    def __init__(self, value):
        super(Dog, self).__init__()
        rate = 1.2
        self.price = self.tax * (rate * value)
```

**Comment by Ovid** [comment]
No. [I go into the reasons why here](https://ovid.github.io/articles/understanding-class-inheritance.html).

**Reply by yuki-kimoto** [reply]
Oh!

I understand the current design and I see your opinion.

However, because the almost major other programming languages are allowed to access the fields of the parent class, we should think about this again. <code>public</code>, <code>private</code>, <code>protected</code>

Do you hope a possible way to access the parent fields?

```
class Point {
  field $x :writer :reader;
  field $y :writer :reader ;
}

class Point3D :isa(Point) {
  method foo () {
    print $self->x . "\n";
  }
}
```
